File: merge_priority_branches.py
#!/usr/bin/env python3
"""
Priority merge script for the most important unmerged branches
"""

import subprocess
import sys
import os

def run_command(cmd, timeout=60):
    """Run a command and return success status and output"""
    try:
        result = subprocess.run(
            cmd, 
            shell=True, 
            capture_output=True, 
            text=True, 
            timeout=timeout
        )
        return result.returncode == 0, result.stdout + result.stderr
    except subprocess.TimeoutExpired:
        return False, "Command timed out"
    except Exception as e:
        return False, f"Error: {e}"

def main():
    print("🔄 Starting priority merge of most important branches...")
    
    # Check if we're on main
    success, output = run_command("git branch --show-current")
    if not success or output.strip() != "main":
        print("❌ Not on main branch. Switching to main...")
        success, output = run_command("git checkout main")
        if not success:
            print(f"❌ Failed to switch to main: {output}")
            return 1
    
    # Pull latest changes
    print("📥 Pulling latest changes...")
    success, output = run_command("git pull origin main")
    if not success:
        print(f"⚠️  Warning: Could not pull latest changes: {output}")
    
    # List of priority branches to merge (most recent and important)
    priority_branches = [
        # Recent content and services (2025)
        "origin/chore/add-more-content-and-promos-2025-09-30",
        "origin/chore/add-oct-13-2025-content-and-home-advertising",
        "origin/chore/add-oct-04-content-and-home-promo",
        "origin/candidate/netlify-build-fix-2025-09-17",
        "origin/candidate/netlify-fix",
        
        # Codex fixes and improvements
        "origin/cecb7u-codex/implement-swr-for-service-fetch",
        "origin/c6z/enhancements",
        
        # Chore improvements
        "origin/chore/add-new-services-and-advertising",
        "origin/chore/add-latest-content-and-promos",
        
        # Build and deployment fixes
        "origin/candidate/build-fix-postcss-esm",
        "origin/candidate/fix-errors-and-merge-to-main-a79e",
        "origin/candidate/fix-syntax-push-and-merge-to-main-44ce",
        
        # TypeScript and component fixes
        "origin/bw06m0-codex/fix-typescript-errors-in-components",
        "origin/bwren1-codex/fix-typescript-errors-in-files",
        
        # Footer and 404 fixes
        "origin/0nylrk-codex-fix-footer-contact-link-resolved",
        "origin/0smfo8-codex-fix-404-error-resolved",
        
        # Build fixes
        "origin/build-fixes-clean",
        "origin/build-fixes-main",
        "origin/build-fixes-working",
        
        # AI dashboard improvements
        "origin/ai-dashboard-improvements",
        
        # Content branches
        "origin/add-new-2026-content",
        "origin/add-revolutionary-content-2026",
        "origin/ai-2027-content-integration"
    ]
    
    successful_merges = []
    failed_merges = []
    skipped_branches = []
    
    for branch in priority_branches:
        print(f"\n🔄 Attempting to merge: {branch}")
        
        # Check if branch exists
        success, output = run_command(f"git ls-remote --heads origin {branch.replace('origin/', '')}")
        if not success or not output.strip():
            print(f"⚠️  Branch {branch} does not exist, skipping...")
            skipped_branches.append(branch)
            continue
        
        # Try to merge
        success, output = run_command(f"git merge {branch} --no-edit --no-ff")
        
        if success:
            print(f"✅ Successfully merged {branch}")
            successful_merges.append(branch)
            
            # Push the merge
            push_success, push_output = run_command("git push origin main")
            if push_success:
                print(f"✅ Pushed merge to origin/main")
            else:
                print(f"⚠️  Could not push to origin: {push_output}")
        else:
            print(f"❌ Failed to merge {branch}")
            print(f"   Error: {output}")
            
            # Try to resolve conflicts automatically
            conflict_files = []
            success, output = run_command("git diff --name-only --diff-filter=U")
            if success and output.strip():
                conflict_files = output.strip().split('\n')
                print(f"🔧 Found conflicts in: {conflict_files}")
                
                # Try to resolve conflicts by taking incoming changes
                for file in conflict_files:
                    if os.path.exists(file):
                        print(f"🔧 Resolving conflicts in {file}...")
                        success, output = run_command(f"git checkout --theirs {file}")
                        if success:
                            run_command(f"git add {file}")
                
                # Try to commit the resolution
                success, output = run_command("git commit --no-edit")
                if success:
                    print(f"✅ Resolved conflicts and merged {branch}")
                    successful_merges.append(branch)
                    
                    # Push the resolved merge
                    push_success, push_output = run_command("git push origin main")
                    if push_success:
                        print(f"✅ Pushed resolved merge to origin/main")
                    else:
                        print(f"⚠️  Could not push to origin: {push_output}")
                else:
                    # Abort the merge
                    run_command("git merge --abort")
                    failed_merges.append(branch)
            else:
                failed_merges.append(branch)
    
    # Summary
    print(f"\n📊 Priority Merge Summary:")
    print(f"✅ Successful merges: {len(successful_merges)}")
    print(f"❌ Failed merges: {len(failed_merges)}")
    print(f"⚠️  Skipped branches: {len(skipped_branches)}")
    
    if successful_merges:
        print(f"\n✅ Successfully merged branches:")
        for branch in successful_merges:
            print(f"   - {branch}")
    
    if failed_merges:
        print(f"\n❌ Failed to merge branches:")
        for branch in failed_merges:
            print(f"   - {branch}")
    
    if skipped_branches:
        print(f"\n⚠️  Skipped branches:")
        for branch in skipped_branches[:5]:  # Show first 5
            print(f"   - {branch}")
        if len(skipped_branches) > 5:
            print(f"   ... and {len(skipped_branches) - 5} more")
    
    # Final status
    print(f"\n📋 Final git status:")
    run_command("git status --short")
    
    print("\n🎉 Priority merge process completed!")
    return 0 if not failed_merges else 1

if __name__ == "__main__":
    sys.exit(main())

File: merge-priority-branches.js
#!/usr/bin/env node
/**
 * Merge Priority Branches - Focus on the most important branches first
 */ import { execSync } from 'child_process';';
import fs from 'fs';'
// //Step 1: Ensure we're on main and up to date,'
// try {,
  execSync('git checkout main', { stdio: 'inherit' })'
  execSync('git pull origin main', { stdio: 'inherit' })'
//   } catch (error) {
  // TODO: Add properties
}
  // TODO: Add properties
}
    //   process.exit(1)
  }

//Step 2: Define priority branches (most recent and important),;
const priorityBranches = [,
// //Step,
  1: Ensure we're on main and up to date,'
// try {/* TODO: Fix JSX expression */}
  o: 'inherit' })'
  execSync('git pull origin main', {/* TODO: Fix JSX expression */})'
  o: 'inherit' })'
//   } catch (error) {/* TODO: Fix JSX expression */}
}

//Step,
  2: Define priority branches (most recent and important);
const priorityBranches = [
  // TODO: Add items
]
  // TODO: Add items
]
  //Recent cursor branches from today,
  'cursor/fix-errors-and-merge-to-main-f4fa','
  'cursor/fix-errors-and-merge-to-main-100d','
  'cursor/fix-errors-and-merge-to-main-9f74','
  'cursor/fix-errors-and-merge-to-main-2f1b','
  'cursor/fix-errors-and-merge-to-main-494d','
  'cursor/fix-errors-and-merge-to-main-7318','
  'cursor/fix-errors-and-merge-to-main-97a6','
  'cursor/fix-errors-and-merge-to-main-9bdd','
  'cursor/fix-errors-and-merge-to-main-a0d2','
  'cursor/fix-errors-and-merge-to-main-d0f7','
  //Content branches
  'add-new-2026-content','
  'add-revolutionary-content-2026','
  'ai-2027-content-integration','
  'ai-dashboard-improvements','
  //Resolved branches
  '0nylrk-codex-fix-footer-contact-link-resolved','
  '0parff-codex-centralize-api-error-handling-resolved','
  '0smfo8-codex-fix-404-error-resolved','
  //Important candidate branches
  'candidate/netlify-build-fix-2025-09-17','
  'candidate/netlify-fix','
  //Important chore branches
  'chore/a11y-reduced-motion','
  'chore/a11y-single-main','
  'chore/add-2027-content-and-homepage-promo','
  'chore/analysis-fixes-and-navigation','
  'chore/animated-section-typing-fix','
  'chore/build-fixes','
  'chore/build-fixes-and-seo','
  'chore/build-fixes-and-slim-app']'
// //Step 3: Merge function with conflict resolution,
function mergeBranch(branchName) {,
//   try {,
    //Check if branch exists
    execSync(`git fetch origin ${branchName}`, { stdio: 'pipe' })'
    //Check if already merged;
const isMerged = execSync(`git branch --merged main | grep -q "${branchName}" || echo "not_merged"`)"
      { encoding: 'utf8' }'
// //Step,
  3: Merge function with conflict resolution,
function mergeBranch(branchName) {/* TODO: Fix JSX expression */}
    execSync(`git fetch origin ${branchName}`, {/* TODO: Fix JSX expression */})
  o: 'pipe' })'
    //Check if already merged;`;
const isMerged = execSync(`git branch --merged main | grep -q "${branchName}" || echo "not_merged"`,"
      {/* TODO: Fix JSX expression */}
  g: 'utf8' })'
    ).trim()
    if (isMerged !== 'not_merged') {/* TODO: Fix JSX expression */}'
  d: 'already_merged' }'
    }

    //Try to merge
    try {
  // TODO: Add properties
}
  // TODO: Add properties
}
      execSync(`git merge origin/${branchName} --no-ff -m "Merge ${branchName}: automated merge"`)"
        { stdio: 'inherit' }'
      )
//       return { success: true, method: 'direct' }'
    } catch (mergeError) {
  // TODO: Add properties
}
  // TODO: Add properties
}
//       //Try different conflict resolution strategies
      try {
  // TODO: Add properties
}
  // TODO: Add properties
}
        //Strategy 1: Use theirs,
        execSync(`git merge origin/${branchName} --strategy-option=theirs --no-ff -m "Merge ${branchName}: using theirs strategy"`)"
          { stdio: 'inherit' }'
        )
//         return { success: true, method: 'theirs' }'
      } catch (theirsError) {
  // TODO: Add properties
}
  // TODO: Add properties
}
        try {
  // TODO: Add properties
}
  // TODO: Add properties
}
          //Strategy 2: Use ours,
          execSync(`git merge origin/${branchName} --strategy-option=ours --no-ff -m "Merge ${branchName}: using ours strategy"`)"
            { stdio: 'inherit' }'
    try {/* TODO: Fix JSX expression */}"`"
        `git merge origin/${branchName} --no-ff -m "Merge ${branchName}: automated merge"`,"
        {/* TODO: Fix JSX expression */}
  o: 'inherit' }'
      )
//       return {/* TODO: Fix JSX expression */}
  d: 'direct' }'
    } catch (mergeError) {/* TODO: Fix JSX expression */}"`"
          `git merge origin/${branchName} --strategy-option=theirs --no-ff -m "Merge ${branchName}: using theirs strategy"`,"
          {/* TODO: Fix JSX expression */}
  o: 'inherit' }'
        )
//         return {/* TODO: Fix JSX expression */}
  d: 'theirs' }'
      } catch (theirsError) {/* TODO: Fix JSX expression */}"`"
            `git merge origin/${branchName} --strategy-option=ours --no-ff -m "Merge ${branchName}: using ours strategy"`,"
            {/* TODO: Fix JSX expression */}
  o: 'inherit' }'
          )
//           return {/* TODO: Fix JSX expression */}
  d: 'ours' }'
        } catch (oursError) {/* TODO: Fix JSX expression */}
  d: 'failed' }'
        }
      }
    }
  } catch (error) {/* TODO: Fix JSX expression */}
  d: 'not_found' }'
  }
}

//Step 4: Process all priority branches;
const results = {
  // TODO: Add properties
}
  // TODO: Add properties
}
  successful: []
  failed: []
  summary: {
  // TODO: Add properties
}
  // TODO: Add properties
}
    total: 0
    successful: 0
    failed: 0
    methods: {
  // TODO: Add properties
}
  // TODO: Add properties
}
      direct: 0
      theirs: 0
      ours: 0
      already_merged: 0
      not_found: 0
      failed: 0,
//Step,
  4: Process all priority branches,;
const results = {/* TODO: Fix JSX expression */}
    }}}

// for (const branch of priorityBranches) {
  // TODO: Add properties
}
  // TODO: Add properties
}
  results.summary.total++
  if (result.success) {
  // TODO: Add properties
}
  // TODO: Add properties
}
    results.successful.push({)
      branch: branch,),
      success: true),
      method: result.method)})
    results.summary.successful++
    results.summary.methods[result.method]++
  } else {
  // TODO: Add properties
}
  // TODO: Add properties
}
    results.failed.push({)
      branch: branch,),
      success: false),
      method: result.method),
// for (const branch of priorityBranches) {/* TODO: Fix JSX expression */}
    })
    results.summary.successful++
    results.summary.methods[result.method]++
  } else {/* TODO: Fix JSX expression */}
    })
    results.summary.failed++
    results.summary.methods[result.method]++
  }
}

//Step 5: Generate report,
// const report = {,
  ...results
  timestamp: new Date().toISOString()}

fs.writeFileSync('priority-branches-merge-report.json')'
  JSON.stringify(report, null, 2)
)
//Step 6: Push changes,
// try {,
  execSync('git push origin main', { stdio: 'inherit' })'
//   } catch (error) {
  // TODO: Add properties
}
  // TODO: Add properties
}
//   }

// Step 7: Summary,
// // // // // // // // // // // // // // ,
//Step,
  5: Generate report,
// const report = {/* TODO: Fix JSX expression */}
}

fs.writeFileSync('priority-branches-merge-report.json')'
  JSON.stringify(report, null, 2)
)
//Step,
  6: Push changes,
// try {/* TODO: Fix JSX expression */}
  o: 'inherit' })'
//   } catch (error) {/* TODO: Fix JSX expression */}
//   }

// Step,
  7: Summary,
// // // // // // // // // // // // // // "`"
